masonite/commands/ServeCommand.py

```diff
@@ -20,20 +18,47 @@ class ServeCommand(Command):
     """
 
     def handle(self):
+        # Check for the 2.0 patch.
         self._check_patch()
+
         if self.option('reload'):
-            self._run_with_reloader(extra_files=[".env"])
+            logger = DefaultLogger(LogLevel.INFO)
+
+            # worker args are pickled and then passed to the new process
+            worker_args = [
+                self.option("host"), self.option("port"), "wsgi:application",
+            ]
+
+            reloader = Reloader(
+                "masonite.commands._devserver.run",
+                find_default_monitor_factory(logger),
+                logger,
+                worker_args=worker_args,
+            )
+
+            self._run_reloader(reloader, extra_files=[".env"])
 
         else:
-            self._run_application()
+            from wsgi import application
+            from ._devserver import run
+            run(self.option("host"), self.option("port"), application)
+
+    def _run_reloader(self, reloader, extra_files=[]):
+        reloader._capture_signals()
+        reloader._start_monitor()
+        for blob in extra_files:
+            reloader.monitor.add_path(os.path.join(os.getcwd(), blob))
+        try:
+            while True:
+                if not reloader._run_worker():
+                    reloader._wait_for_changes()
+                time.sleep(reloader.reload_interval)
```

Comment:
I think i understand what this code is doing: giving time for the PID to be killed before starting a new one?

but heres a crazy idea: if we comment this line of code out, the reloader speed increases by like 23x. Thats more than an order of magnitude faster. See the screen shots attached. With it there the reloader is 92 microseconds but with it commented out it is around 4-5 micro seconds

<img width="785" alt="screen shot 2018-11-13 at 11 41 02 pm" src="https://user-images.githubusercontent.com/20172538/48460631-ec96fd00-e79d-11e8-86c2-a973ad16e4a8.png">
<img width="748" alt="screen shot 2018-11-13 at 11 41 27 pm" src="https://user-images.githubusercontent.com/20172538/48460632-ec96fd00-e79d-11e8-8703-567ebb82754b.png">

Reply:
Now what I THINK i have noticed though is that it does indeed not kill a process very very occasionally. and even when it doesn't LOOK like it kills the process, i'm not sure it doesn't. Take these snippets for example. If i save and edit SUPER fast. Like 3 saves per seconds going crazy in order to mock the starting and killing of proccesses very fast I can get the process to start but never kill in the terminal. 

BUT i can never get more then 2 processes in activity monitor. So i'm not sure it doesn't kill it. This is with the sleep line of code commented out.

<img width="722" alt="screen shot 2018-11-13 at 11 52 37 pm" src="https://user-images.githubusercontent.com/20172538/48460996-8a3efc00-e79f-11e8-8924-08fe8d88ba7f.png">
<img width="742" alt="screen shot 2018-11-13 at 11 51 09 pm" src="https://user-images.githubusercontent.com/20172538/48461011-9d51cc00-e79f-11e8-803c-e044d91e2805.png">

Reply:
actually relooking at it it does actually kill it right below it. I didn't see that. I honestly think we can just uncomment that sleep code and speed this up by ~25x